### PR TITLE
Fix for #5898 - Converting native ETH to fiat and fiat to native ETH results in wrong values beign displayed on the Amount screen

### LINKED ISSUE #5898: [Bug] Converting native ETH to fiat and fiat to native ETH results in wrong values beign displayed on the Amount screen
**Describe the bug**
When I am performing a Send ETH and I introduce an amount, if I click to convert to fiat and back to native, the amounts are not correct anymore.

**Screenshots**

https://user-images.githubusercontent.com/54408225/223138506-62560219-f391-4431-86c2-9e8b41bedf70.mp4

**To Reproduce**
_Steps to reproduce the behavior_
1. Click Send ETH
2. Add Recipient
3. Introduce amount
4. Switch to fiat -- see incorrect amount
5. Switch to native -- see incorrect amount

**Expected behavior**
Conversion values should be correct

**Smartphone (please complete the following information):**

- 
- Device: Pixel 6
- OS: Android 12
- App Version: v6.0.1 (1078)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by segun**
**Description**

This PR fixes #5898, a bug that shows wrong results during etc/fiat conversion

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/SendFlow/Amount/index.js [hunk 1]
```diff
@@ -1122,10 +1123,14 @@ class Amount extends PureComponent {
...
-    });
-    this.onInputChange(inputValueConversion);
+    this.setState(
+      {
+        internalPrimaryCurrencyIsCrypto: !internalPrimaryCurrencyIsCrypto,
+      },
+      () => {
+        this.onInputChange(inputValueConversion);
+      },
+    );
```
Comment (jpuri) [inline app/components/Views/SendFlow/Amount/index.js:1133]: 👍

**Review by jpuri — APPROVED** [review]

**Review by blackdevelopa — APPROVED** [review]

**Comment by bschorchit** [comment]
Has this been QAed? @seaona

**Comment by seaona** [comment]
@bschorchit  this has not been QAd.
@segun notice that after having 2 dev reviews, the developer should add the `needs-qa` label, so the QA engineer can test it. Once QA has passed, the QA engineer adds the label `QA-passed` and merges the PR to main. Finally a release label is added

**Comment by seaona** [comment]
I can see how the issue is fixed. I add the QA Pass label.

- Conversion is correctly performed on Mainnet

[Screencast from 13-04-23 14:28:26.webm](https://user-images.githubusercontent.com/54408225/231758417-83e81cc4-9bdd-46cb-817e-bef080ac064a.webm)

- Conversion is correctly performed on another networks

[Screencast from 13-04-23 14:29:31.webm](https://user-images.githubusercontent.com/54408225/231758579-cdf79121-33cf-4b1e-b072-ebf223a76c82.webm)

- Conversion is correctly performed with ERC20 tokens

[Screencast from 13-04-23 14:32:09.webm](https://user-images.githubusercontent.com/54408225/231759776-6e6f6a6f-d4c6-4d75-904b-e270bfd96403.webm)

**Pull request merged**